packages/core/src/core.ts

```diff
@@ -182,13 +183,8 @@ export class Core extends EventEmitter<CoreEvents> {
 	/**
 	 * Helper for verifying the blockhash for an action or session.
 	 */
-	public async verifyBlock(blockInfo: Block) {
-		const { chain, chainId, blocknum, blockhash, timestamp } = blockInfo
-		const block = await this.blockResolver(chain, chainId, blockhash)
-
-		// check the block retrieved from RPC matches metadata from the user
-		assert(block.number === blocknum, "action/session provided with invalid block number")
-		assert(block.timestamp === timestamp, "action/session provided with invalid timestamp")
+	public async verifyBlock({ chain, chainId, blockhash }: { chain: Chain; chainId: ChainId; blockhash: string }) {
+		await this.blockResolver(chain, chainId, blockhash)
```

Comment:
we should pass payload.timestamp into here and assert that it's after the block's timestamp % a small grace period